moveit_experimental/jog_arm/src/jog_arm/jog_arm_server.cpp

```diff
@@ -0,0 +1,1254 @@
+    outgoing_cmd_pub = n.advertise<std_msgs::Float64MultiArray>(ros_parameters_.command_out_topic, 1);
+
+  // Wait for low pass filters to stabilize
+  ros::Duration(10 * ros_parameters_.publish_period).sleep();
+
+  ros::Rate main_rate(1. / ros_parameters_.publish_period);
+
+  while (ros::ok())
+  {
+    ros::spinOnce();
+
+    pthread_mutex_lock(&shared_variables_.new_traj_mutex);
+    trajectory_msgs::JointTrajectory new_traj = shared_variables_.new_traj;
+    pthread_mutex_unlock(&shared_variables_.new_traj_mutex);
+
+    // Check for stale cmds
+    pthread_mutex_lock(&shared_variables_.incoming_cmd_stamp_mutex);
+    if ((ros::Time::now() - shared_variables_.incoming_cmd_stamp) <
+        ros::Duration(ros_parameters_.incoming_command_timeout))
+    {
+      // Mark that incoming commands are not stale
+      pthread_mutex_lock(&shared_variables_.command_is_stale_mutex);
+      shared_variables_.command_is_stale = false;
+      pthread_mutex_unlock(&shared_variables_.command_is_stale_mutex);
+    }
+    else
+    {
+      pthread_mutex_lock(&shared_variables_.command_is_stale_mutex);
+      shared_variables_.command_is_stale = true;
+      pthread_mutex_unlock(&shared_variables_.command_is_stale_mutex);
+    }
+    pthread_mutex_unlock(&shared_variables_.incoming_cmd_stamp_mutex);
+
+    // Publish the most recent trajectory, unless the jogging calculation thread
+    // tells not to
+    pthread_mutex_lock(&shared_variables_.ok_to_publish_mutex);
+    if (shared_variables_.ok_to_publish)
+    {
+      // Put the outgoing msg in the right format
+      // (trajectory_msgs/JointTrajectory or std_msgs/Float64MultiArray).
+      if (ros_parameters_.command_out_type == "trajectory_msgs/JointTrajectory")
+      {
+        new_traj.header.stamp = ros::Time::now();
+        outgoing_cmd_pub.publish(new_traj);
+      }
+      else if (ros_parameters_.command_out_type == "std_msgs/Float64MultiArray")
+      {
+        std_msgs::Float64MultiArray joints;
+        if (ros_parameters_.publish_joint_positions)
+          joints.data = new_traj.points[0].positions;
+        else if (ros_parameters_.publish_joint_velocities)
+          joints.data = new_traj.points[0].velocities;
+        outgoing_cmd_pub.publish(joints);
+      }
+    }
+    else
+    {
+      ROS_WARN_STREAM_THROTTLE_NAMED(2, NODE_NAME, "Stale or zero command. "
+                                                   "Try a larger 'incoming_command_timeout' parameter?");
+    }
+    pthread_mutex_unlock(&shared_variables_.ok_to_publish_mutex);
+
+    main_rate.sleep();
+  }
+
+  (void)pthread_join(joggingThread, nullptr);
+  (void)pthread_join(collisionThread, nullptr);
+}
+
+// A separate thread for the heavy jogging calculations.
+void* JogROSInterface::jogCalcThread(void*)
+{
+  jog_arm::JogCalcs ja(ros_parameters_, shared_variables_, model_loader_ptr_);
+  return nullptr;
+}
+
+// A separate thread for collision checking.
+void* JogROSInterface::collisionCheckThread(void*)
```

Comment:
```suggestion
void* JogROSInterface::collisionCheckThread()
```

passing in void is C-style

Reply:
From the examples I see online, this is how pthreads are usually used. Anyway, I get compile errors if I try to eliminate that argument.